src/pipeline/base_pipeline.py

```diff
@@ -248,5 +250,13 @@ def _print_and_write_log(self, epoch, worker_outputs, write=True):
                 if global_config.verbosity >= 1:
                     logger.info(f'    {str(key):20s}: {value:.4f}')
                 if 'elapsed_time' not in key and write:
+                    value = value.item() if isinstance(value, torch.Tensor) else value
+                    epoch_record[f'{loader_name}_{key}'] = [value]
                     # TODO: See if we can use tree-structured tensorboard logging
                     self.writer.add_scalar(f'{loader_name}_{key}', value)
+
```

Comment:
We may also want to update the comments for the `_print_and_write_log` function

Reply:
Updated